src/Operation/Aggregate.php

```diff
@@ -295,7 +294,7 @@ public function execute(Server $server)
 
         $result = current($cursor->toArray());
 
-        if (! isset($result->result) || ! is_array($result->result)) {
+        if (! is_object($result) || ! isset($result->result) || ! is_array($result->result)) {
```

Comment:
Does psalm complain about using a property accessor on a potential non-object even if PHP would ignore it thanks to `isset()`?

Reply:
That is correct, here the corresponding psalm error:

ERROR: PossiblyInvalidPropertyFetch - src/Operation/Aggregate.php:297:22 - Cannot fetch property on possible non-object `$result` of type `array<array-key, mixed>` (see https://psalm.dev/114)
```
        if ( ! isset($result->result) || ! is_array($result->result)) {
```

I've asked to figure out whether this is a bug or intended behaviour. Either way, this issue goes away once we properly handle the type map in our methods.